### ISSUE TITLE
AMQP 2.1.4 fails on FreeBSD 9.2 with OperationError: [Errno 42] Protocol not available

### ISSUE THREAD

**Issue by dhebvvs** [celery/py-amqp#124]
Versions of packages involved:
- Python 2.7.9
- celery 4.0.2
- kombu 4.0.2
- amqp 2.1.4

As a temporary fix we have downgraded to amqp 2.1.3, which seems to still work.

See: https://github.com/celery/py-amqp/blob/v2.1.3/amqp/transport.py#L194

We use the `pyamqp://`-protoco in Celeryl, the RabbitMQ server is running version 3.6.6.

Stacktrace:

```
  File ".../site-packages/celery/app/base.py", line 737, in send_task
    amqp.send_task_message(P, name, message, **options)
  File "/usr/local/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File ".../site-packages/kombu/connection.py", line 419, in _reraise_as_library_errors
    sys.exc_info()[2])
  File ".../site-packages/kombu/connection.py", line 414, in _reraise_as_library_errors
    yield
  File ".../site-packages/celery/app/base.py", line 736, in send_task
    self.backend.on_task_call(P, task_id)
  File ".../site-packages/celery/backends/rpc.py", line 168, in on_task_call
    maybe_declare(self.binding(producer.channel), retry=True)
  File ".../site-packages/kombu/messaging.py", line 209, in _get_channel
    channel = self._channel = channel()
  File ".../site-packages/kombu/utils/functional.py", line 38, in __call__
    value = self.__value__ = self.__contract__()
  File ".../site-packages/kombu/messaging.py", line 224, in <lambda>
    channel = ChannelPromise(lambda: connection.default_channel)
  File ".../site-packages/kombu/connection.py", line 819, in default_channel
    self.connection
  File ".../site-packages/kombu/connection.py", line 802, in connection
    self._connection = self._establish_connection()
  File ".../site-packages/kombu/connection.py", line 757, in _establish_connection
    conn = self.transport.establish_connection()
  File ".../site-packages/kombu/transport/pyamqp.py", line 130, in establish_connection
    conn.connect()
  File ".../site-packages/amqp/connection.py", line 294, in connect
    self.transport.connect()
  File ".../site-packages/amqp/transport.py", line 122, in connect
    self.socket_settings, self.read_timeout, self.write_timeout,
  File ".../site-packages/amqp/transport.py", line 174, in _init_socket
    self._set_socket_options(socket_settings)
  File ".../site-packages/amqp/transport.py", line 198, in _set_socket_options
    tcp_opts = self._get_tcp_socket_defaults(self.sock)
  File ".../site-packages/amqp/transport.py", line 194, in _get_tcp_socket_defaults
    opt: sock.getsockopt(SOL_TCP, opt) for opt in TCP_OPTS
  File ".../site-packages/amqp/transport.py", line 194, in <dictcomp>
    opt: sock.getsockopt(SOL_TCP, opt) for opt in TCP_OPTS
  File "/usr/local/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
OperationalError: [Errno 42] Protocol not available
```

**Comment by eugene-s** [comment]
Same error in Bash on Windows.
Windows 10 Build 1703.

Versions of packages involved:

- Python 3.6.1
- kombu 4.0.2
- amqp 2.1.2 and more

I had to downgrade to version 2.1.1. It works for me.

`  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/connection.py", line 419, in _reraise_as_library_errors
    sys.exc_info()[2])
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/vine/five.py", line 175, in reraise
    raise value.with_traceback(tb)
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/connection.py", line 414, in _reraise_as_library_errors
    yield
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/connection.py", line 405, in ensure_connection
    callback)
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/utils/functional.py", line 333, in retry_over_time
    return fun(*args, **kwargs)
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/connection.py", line 261, in connect
    return self.connection
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/connection.py", line 802, in connection
    self._connection = self._establish_connection()
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/connection.py", line 757, in _establish_connection
    conn = self.transport.establish_connection()
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/kombu/transport/pyamqp.py", line 130, in establish_connection
    conn.connect()
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/amqp/connection.py", line 294, in connect
    self.transport.connect()
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/amqp/transport.py", line 122, in connect
    self.socket_settings, self.read_timeout, self.write_timeout,
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/amqp/transport.py", line 174, in _init_socket
    self._set_socket_options(socket_settings)
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/amqp/transport.py", line 198, in _set_socket_options
    tcp_opts = self._get_tcp_socket_defaults(self.sock)
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/amqp/transport.py", line 194, in _get_tcp_socket_defaults
    opt: sock.getsockopt(SOL_TCP, opt) for opt in TCP_OPTS
  File "/root/.virtualenvs/project/lib/python3.6/site-packages/amqp/transport.py", line 194, in <dictcomp>
    opt: sock.getsockopt(SOL_TCP, opt) for opt in TCP_OPTS
kombu.exceptions.OperationalError: [Errno 92] Protocol not available`

**Comment by thedrow** [comment]
This is now fixed by #211.
